src/js/factory/weekView.js

```diff
@@ -31,52 +31,92 @@ var TimeResize = require('../handler/time/resize');
 var MilestoneClick = require('../handler/milestone/click');
 var TaskClick = require('../handler/task/click');
 
+var DEFAULT_VIEW_SEQUENCE = ['Milestone', 'Task', 'AllDay', 'TimeGrid'];
+var DEFAULT_VIEWS = {
+    'Milestone': {
+        minHeight: 20,
+        height: 80,
+        maxHeight: 80,
+        show: true
+    },
+    'Task': {
+        minHeight: 40,
+        height: 120,
+        maxHeight: 120,
+        show: true
+    },
+    'AllDay': {
+        minHeight: 20,
+        height: 80,
+        maxHeight: 80,
+        show: true
+    },
+    'TimeGrid': {
+        autoHeight: true,
+        show: true
+    }
+};
+
+/* eslint-disable complexity*/
 module.exports = function(baseController, layoutContainer, dragHandler, options) {
     var weekView, dayNameContainer, dayNameView, vLayoutContainer, vLayout,
         milestoneView, taskView, alldayView, timeGridView,
-        milestoneClickHandler, taskClickHandler, alldayClickHandler, alldayCreationHandler,
-        alldayMoveHandler, alldayResizeHandler, timeClickHandler, timeCreationHandler,
-        timeMoveHandler, timeResizeHandler, daynameClickHandler,
-        panels;
+        panels = [];
+    var viewSequence = options.week.viewSequence || DEFAULT_VIEW_SEQUENCE,
+        views = options.week.views || DEFAULT_VIEWS;
 
     weekView = new Week(null, options.week, layoutContainer);
+    weekView.handler = {
+        click: {},
+        dayname: {},
+        creation: {},
+        move: {},
+        resize: {}
+    };
+
+    // Change visibilities
+    util.forEach(views, function(value, key) {
+        if (key === 'Milestone' || key === 'Task') {
+            value.show = options.taskView;
+        } else if (key === 'AllDay' || key === 'TimeGrid') {
+            value.show = options.scheduleView;
+        }
+    });
+
+    // Make panels by view sequence and visibilities
+    util.forEach(viewSequence, function(name) {
+        var view = views[name];
+        if (view.show) {
+            if (panels.length) {
+                panels.push({
+                    isSplitter: true
+                });
+            }
+            panels.push(util.extend({
+                name: name
+            }, view));
+        }
+    });
+
+    if (panels.length) {
+        panels[panels.length - 1].autoHeight = true;
+        panels[panels.length - 1].maxHeight = null;
+    }
+
     dayNameContainer = domutil.appendHTMLElement('div', weekView.container, config.classname('dayname-layout'));
 
     /**********
      * 일자표기 (상단 일월화수...)
      **********/
     dayNameView = new DayName(options.week, dayNameContainer);
-    daynameClickHandler = new DayNameClick(dragHandler, dayNameView, baseController);
+    weekView.handler.dayname.date = new DayNameClick(dragHandler, dayNameView, baseController);
```

Comment:
I think property name should be `weekView.handler.click.dayname`.
Other handler name which is initiated by ##Click constructor is weekView.handler.click.##.

Reply:
Yes, I agree. I should be renamed in next issue.
I have a refactoring plan, so it will be changed.